pkg/utils/predicates.go

```diff
@@ -322,11 +322,21 @@ func CheckPodsWithAntiAffinityExist(pod *v1.Pod, pods map[string][]*v1.Pod, node
 			}
 			for namespace := range namespaces {
 				for _, existingPod := range pods[namespace] {
-					if existingPod.Name != pod.Name && podMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector) {
-						node, ok := nodeMap[pod.Spec.NodeName]
-						if !ok {
-							continue
+					if existingPod.Name == pod.Name || !podMatchesTermsNamespaceAndSelector(existingPod, namespaces, selector) {
+						continue
+					}
+
+					node, ok := nodeMap[pod.Spec.NodeName]
```

Comment:
There is [only one Node being passed](https://github.com/kubernetes-sigs/descheduler/pull/1395/files#diff-fc4033754cd376d78b09ff84a934296f26e774702505885309268f816ba3817cR348) when coming from NodeFit and that is the destination Node (N2). P1 is not on N2, so this will be `!ok`. We should only evaluate this if `nodeFit == false`.

Reply:
thank you, should using existingPod here, i have fixed it.

Reply:
Perhaps my pr misled ingvagabund,😂

Reply:
> should using existingPod here

Yes! I agree about reversing to make sure that existingPod is on N2 👍